test/test_producer.py

```diff
@@ -3,6 +3,7 @@
 import logging
 
 from mock import MagicMock
+from multiprocessing.queues import Full
```

Comment:
In `base.py`, you have `from queue import Full`, or `from Queue import Full`. Would stick with that.